sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,261 @@
+    KBinsDiscretizer(...)
+    >>> Xt = est.transform(X)
+    >>> Xt  # doctest: +SKIP
+    array([[ 0., 0., 0., 0.],
+           [ 1., 1., 1., 0.],
+           [ 2., 2., 2., 1.],
+           [ 2., 2., 2., 2.]])
+
+    Sometimes it may be useful to convert the data back into the original
+    feature space. The ``inverse_transform`` function converts the binned
+    data into the original feature space. Each value will be a distance
+    of ``bin_width_ / 2`` from the nearest bin edge.
+
+    >>> est.bin_width_
+    array([ 1.,  1.,  1.,  1.])
+    >>> est.inverse_transform(Xt)
+    array([[-1.5,  1.5, -3.5, -0.5],
+           [-0.5,  2.5, -2.5, -0.5],
+           [ 0.5,  3.5, -1.5,  0.5],
+           [ 0.5,  3.5, -1.5,  1.5]])
+
+    Notes
+    -----
+    Bin edges for feature ``i`` are defined as::
+
+      np.concatenate([
+        -np.inf,
+        offset_[i] + bin_width_[i] * np.arange(1, n_bins_[i]),
+        np.inf
+      ])
+
+    See also
+    --------
+     sklearn.preprocessing.Binarizer : class used to bin values as ``0`` or
+        ``1`` based on a parameter ``threshold``.
+    """
+
+    def __init__(self, n_bins=2, ignored_features=None):
+        self.n_bins = n_bins
+        self.ignored_features = ignored_features
+
+    def fit(self, X, y=None):
+        """Fits the estimator.
+
+        Parameters
+        ----------
+        X : numeric array-like, shape (n_samples, n_features)
+            Data to be discretized.
+
+        y : ignored
+
+        Returns
+        -------
+        self
+        """
+        X = check_array(X, dtype=FLOAT_DTYPES)
+
+        n_features = X.shape[1]
+        ignored = self._check_ignored_features(self.ignored_features,
+                                               n_features)
+        self.transformed_features_ = np.delete(np.arange(n_features), ignored)
+
+        offset = np.min(X, axis=0)
+        offset[ignored] = 0
+        self.offset_ = offset
+
+        n_bins = self._check_n_bins(self.n_bins, n_features, ignored)
+        n_bins[ignored] = 0
+        self.n_bins_ = n_bins
+
+        ptp = np.ptp(X, axis=0)
+        same_min_max = np.where(ptp == 0)[0]
+        if len(same_min_max) > 0:
+            warnings.warn("Features {} are constant and will be replaced with "
+                          "0.".format(", ".join(str(i) for i in same_min_max)))
+
+        with np.errstate(divide='ignore', invalid='ignore'):
+            bin_widths = ptp / self.n_bins_
+        bin_widths[ignored] = 0
+        self.bin_width_ = bin_widths
+        return self
+
+    @staticmethod
+    def _check_n_bins(n_bins, n_features, ignored):
+        if isinstance(n_bins, numbers.Number):
+            if n_bins < 2:
+                raise ValueError("Discretizer received an invalid number "
+                                 "of bins. Received {0}, expected at least 2."
+                                 .format(n_bins))
+            return np.ones(n_features) * n_bins
```

Comment:
I am not sure to understand. What I mean is the following:

```python
np.ones(n_features, dtype=np.int) * n_bins
```
Moreover if the previous `isinstance` is done with `Integral` you are ensuring that the type of self.n_bins_ will always be integral (which is should be since it is just a count)